plugins/linuxplugin/ifplugin/interface_config.go

```diff
@@ -197,328 +207,506 @@ func (plugin *LinuxInterfaceConfigurator) LookupLinuxInterfaces() error {
+			ifConfig.config.Name, ifConfig.config.HostIfName)
+		return nil
+	}
+
+	// Try to find interface in default namespace
+	var found bool
+	for _, linuxIf := range linuxIfs {
+		if ifConfig.config.HostIfName == linuxIf.Attrs().Name {
+			if linuxIf.Type() == tapTun {
+				found = true
+				break
+			}
+			plugin.Log.Debugf("Linux TAP config %v found linux interface %v, but it is not the TAP interface type",
+				ifConfig.config.Name, ifConfig.config.HostIfName)
+		}
+	}
+	if !found {
+		plugin.Log.Debugf("Linux TAP config %v did not found the linux interface with name %v", ifConfig.config.Name,
+			ifConfig.config.HostIfName)
+		return nil
+	}
+
+	return plugin.configureLinuxInterface(nsMgmtCtx, ifConfig.config)
+}
+
+// Set linux interface to proper namespace and configure attributes
+func (plugin *LinuxInterfaceConfigurator) configureLinuxInterface(nsMgmtCtx *linuxcalls.NamespaceMgmtCtx, ifConfig *interfaces.LinuxInterfaces_Interface) (err error) {
+	if ifConfig.HostIfName == "" {
+		return fmt.Errorf("host interface not specified for %v", ifConfig.Name)
 	}
-	// Push defer to a stack as the first one, so it will be called last
-	defer revertCfgNs()
 
 	// Move interface to the proper namespace.
-	ns := iface.config.Namespace
+	ns := ifConfig.Namespace
 	if ns != nil && ns.Type == interfaces.LinuxInterfaces_Interface_Namespace_MICROSERVICE_REF_NS {
 		ns = plugin.convertMicroserviceNsToPidNs(ns.Microservice)
 		if ns == nil {
 			return &unavailableMicroserviceErr{}
 		}
 	}
-	err = linuxcalls.SetInterfaceNamespace(nsMgmtCtx, iface.config.HostIfName, ns, plugin.Log, plugin.Stopwatch)
+
+	err = linuxcalls.SetInterfaceNamespace(nsMgmtCtx, ifConfig.HostIfName, ifConfig.Namespace, plugin.Log, plugin.Stopwatch)
 	if err != nil {
-		return fmt.Errorf("failed to move interface across namespaces: %v", err)
+		return fmt.Errorf("failed to set interface %s to namespace %s: %v", ifConfig.HostIfName, ifConfig.Namespace, err)
 	}
 
 	// Continue configuring interface in its namespace.
-	revertNs, err := plugin.switchToNamespace(nsMgmtCtx, iface.config.Namespace)
+	revertNs, err := plugin.switchToNamespace(nsMgmtCtx, ifConfig.Namespace)
 	if err != nil {
 		return fmt.Errorf("failed to switch network namespace: %v", err)
 	}
 	defer revertNs()
 
+	var wasErr error
+
 	// Set interface up.
-	if iface.config.Enabled {
-		err := linuxcalls.InterfaceAdminUp(iface.config.HostIfName, measure.GetTimeLog("iface_admin_up", plugin.Stopwatch))
+	if ifConfig.Enabled {
+		err := linuxcalls.InterfaceAdminUp(ifConfig.HostIfName, measure.GetTimeLog("iface_admin_up", plugin.Stopwatch))
 		if nil != err {
-			return fmt.Errorf("failed to enable Linux interface: %v", err)
+			wasErr = fmt.Errorf("failed to enable Linux interface: %v", err)
+			plugin.Log.Error(wasErr)
 		}
 	}
 
-	var wasError error
-
-	// Configure optional mac address.
-	if iface.config.PhysAddress != "" {
```

Comment:
Now the mac is set even if it's empty? Does this force clients to set some mac for veth/tap interfaces?